cypress/support/index.js

```diff
@@ -12,9 +12,33 @@
 // You can read more here:
 // https://on.cypress.io/configuration
 // ***********************************************************
+import addContext from 'mochawesome/addContext'
 
 // Import commands.js using ES2015 syntax:
 import './commands'
 
 // Alternatively you can use CommonJS syntax:
 // require('./commands')
+
+Cypress.on('test:after:run', (test, runnable) => {
+
+  if (test.state === 'failed') {
+    let item = runnable
+    const nameParts = [runnable.title]
+
+    while (item.parent) {
+      nameParts.unshift(item.parent.title)
+      item = item.parent
+    }
+
+    const fullTestName = nameParts
```

Comment:
When tests will fail on hook Cypress add such info to the screenshot name, in example "before all hook". In such case it should be also added to the name for example by adding 
`if(runnable.hookName) {
            nameParts.push(``${runnable.hookName} hook``)
        }`